scipy/stats/_distn_infrastructure.py

```diff
@@ -1451,17 +1451,30 @@ def support(self, *args, **kwargs):
         args, loc, scale = self._parse_args(*args, **kwargs)
         arrs = np.broadcast_arrays(*args, loc, scale)
         args, loc, scale = arrs[:-2], arrs[-2], arrs[-1]
-        cond = self._argcheck(*args) & (scale > 0)
+        cond = self._argcheck(*args) & (scale > 0) & (loc == loc)
```

Comment:
What is the point of doing `& (loc == loc)`?

Reply:
> What is the point of doing `& (loc == loc)`?

It is used to mask the `nan` values:

```py
>>> import numpy as np
>>> 
>>> x = np.array([1, 2, np.nan, np.inf, -np.inf, np.nan])
>>> x
array([  1.,   2.,  nan,  inf, -inf,  nan])
>>> x == x
array([ True,  True, False,  True,  True, False])
>>>
```

Reply:
wouldn't it be cleaner like this?
```python
In [2]: import numpy as np

In [3]: x = np.array([1, 2, np.nan, np.inf, -np.inf, np.nan])

In [4]: x
Out[4]: array([  1.,   2.,  nan,  inf, -inf,  nan])

In [5]:

In [5]: ~np.isnan(x)
Out[5]: array([ True,  True, False,  True,  True, False])
```

Reply:
```suggestion
        cond = self._argcheck(*args) & (scale > 0) & ~np.isnan(loc)
```

Reply:
I would be. I have changed it now. It's just that historically SciPy has preferred to write `~np.isnan(loc)` as `loc == loc`. But I am OK with both.